backend/015-favorite-meal-plan.sql

```diff
@@ -0,0 +1,30 @@
+-- Create the favorite_meal_plan table if it doesn't exist
+CREATE TABLE IF NOT EXISTS app.favorite_meal_plan (
```

Comment:
This should just be called favorite meals. Ideally, you want answers for this. 

This is to create a favorite meal. The current user logged in should be retrieved from the cookie which can be gotten from `current_settings`.

```
mutation {
  favoriteMeal(mealId: "id") {
    // this returns meal
  }
}
```

For the query, you need to get meals for a logged in user.

```
query {
  favoriteMeals {
    // attributes for the meal.
  }
}
```

For the sake of completeness we also need to unfavorite a meal.

```
mutation {
  unfavoriteMeal(mealId: "1") {
    // attributes of the meal.
  }
}
```

We need to create two functions for creating and removing the meal. We also need another function to return the favoriteMeal of the current user. 

Potentially a meal designer can know how favorite a meal is by retrieving the users who have favoritied a meal.

```
query {
  meal(id: "id") {
    // other attributes
    favoritedBy {
      nodes {
          fullName
          email
           id
       }
    }
  }
}
```